### PR TITLE
feat(frontend): allow running task cancellation

### LINKED ISSUE #6254: Introduce the ability to cancel async queries
## Abstract

We need the endpoint `DELETE /api/1/tasks/{task_id}` to provide the ability to terminate an in-progress `async_query`.

Integrate this functionality with the pending tasks in the notification area so you can stop the execution of a pending task.

## Motivation

Sometimes you might need to cancel long-running tasks for queries that have been rate-limited for hours.

## Specification

- `DELETE /api/1/tasks/{task_id}`
- Frontend integration

### PR DESCRIPTION
**Pull request by tewshi**
Closes #6254 

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/locales/en.json [hunk 1]
```diff
@@ -1518,7 +1518,8 @@
     "title": "Exchange Trades"
   },
   "collapsed_pending_tasks": {
-    "title": "{count} pending task | {count} pending tasks"
+    "title": "{count} pending task | {count} pending tasks",
+    "cancel_task": "cancel task"
```
Comment (lukicenturi) [inline frontend/app/src/locales/en.json:1522]: "Cancel task"

**Comment by lukicenturi** [comment]
I think a confirmation dialog is needed when you click the cancel task button.

**Comment by lukicenturi** [comment]
I wonder if we somehow can unify the implementation for the `UserCancelledTaskError` in catch.
Since the only thing it does is `logger.debug`.

File: frontend/app/src/store/tasks/index.ts [hunk 2]
```diff
@@ -169,6 +190,18 @@ export const useTaskStore = defineStore('tasks', () => {
           } else if (result === null) {
             let errorMessage: string;
             if (message) {
+              if (message === USER_CANCELLED_TASK) {
+                if (checkIfDevelopment() && !import.meta.env.VITE_TEST) {
```
Comment (lukicenturi) [inline frontend/app/src/store/tasks/index.ts:194]: maybe put the logger here.
and in other task caller, in the `catch part`, just do:

`if (!(e instanceof UserCancelledTaskError)) {...`

**Comment by tewshi** [comment]
> I think a confirmation dialog is needed when you click the cancel task button.

okay, should do.

File: frontend/app/src/store/tasks/index.ts (hunk 2 above)
Comment (tewshi) [reply frontend/app/src/store/tasks/index.ts:194]: okay, I see, makes sense.

File: frontend/app/src/composables/assets/retrieval.ts [hunk 3]
```diff
@@ -154,13 +154,15 @@ export const useAssetInfoRetrieval = () => {
...
       return result;
     } catch (e: any) {
-      notify({
-        title: t('actions.assets.erc20.error.title', payload),
-        message: t('actions.assets.erc20.error.description', {
-          message: e.message
-        }),
-        display: true
-      });
+      if (!(e instanceof UserCancelledTaskError)) {
```
Comment (lukicenturi) [inline frontend/app/src/composables/assets/retrieval.ts:157]: After I think again, is it makes sense to move this to a function instead?
Something like `processError` that returns boolean

Comment (lukicenturi) [reply frontend/app/src/composables/assets/retrieval.ts:157]: and also if possible, use return early.
`if(!processError(e)) return`

Comment (tewshi) [reply frontend/app/src/composables/assets/retrieval.ts]: done

**Review by lukicenturi — APPROVED** [review]
LGTM!

**Pull request merged**